packages/python/plotly/plotly/io/_utils.py

```diff
@@ -1,6 +1,29 @@
 import plotly
 import plotly.graph_objs as go
 from plotly.offline import get_plotlyjs_version
+from plotly import optional_imports
+
+import warnings
+import psutil
+
+jupyter_notebook = optional_imports.get_module("notebook")
+jupyter_lab = optional_imports.get_module("jupyterlab")
+
+
+def display_jupyter_version_warnings():
+
+    parent_process = psutil.Process().parent().cmdline()[-1]
+
+    if "jupyter-notebook" in parent_process and jupyter_notebook.__version__ < "7":
```

Comment:
Can we move the `optional_imports.get_module` calls inside these conditionals? Something like:
```py
if "jupyter-notebook" in parent_process:
    jupyter_notebook = optional_imports.get_module("notebook")
    if jupyter_notebook.__version__ < "7":
```
and similar for `jupyterlab`. Those imports can be slow (particularly `jupyterlab`, it's nearly a second on my computer), no need to increase everyone's startup time (for example Dash apps). And calling `optional_imports.get_module` on an already-loaded module or a nonexistent module you've previously asked for is very fast.

I wonder about the existing optional import calls in `_renderers.py` as well - looks like `IPython` and `IPython.display` are automatically imported when you're in an IPython or jupyter environment, so I bet we could just add `should_load=False` to those calls. `nbformat` is not automatically loaded, so it would need to move into the `show` method where it's called, like I'm suggesting here.

Reply:
Good point! I moved them into the if statements. I'm not sure about the other optional calls in that file but I can make an issue to investigate that more

Reply:
Just made this issue: https://github.com/plotly/plotly.py/issues/4856